runtime/core/decoder/torch_asr_decoder.cc

```diff
@@ -120,8 +124,8 @@ bool TorchAsrDecoder::AdvanceDecoding() {
                                       ->run_method("ctc_activation", chunk_out)
                                       .toTensor()[0];
     encoder_outs_.push_back(std::move(chunk_out));
-    ctc_prefix_beam_searcher_->Search(ctc_log_probs);
-    UpdateResult();
+    UpdateResult(ctc_log_probs);
+    finish = IsEndpoint(ctc_log_probs) || finish;
 
```

Comment:
It's better if we can create a class named CtcEndpointer, which has it's own state related to endpoint, and accept ctc_log_probs as input, and it has the function like EndpointDetected(int frame), just like what we do in recognizer.